File: flutter/sumisid_client/lib/models/club.dart
class Club {
  final int id;
  final String name;
  final String streetAddress;
  final String streetNumber;
  final String zipCode;
  final String phoneNumber;
  final int contactPersonId;
  final String extraInfo;

  const Club(
      {required this.id,
      required this.name,
      required this.streetAddress,
      required this.streetNumber,
      required this.zipCode,
      required this.phoneNumber,
      required this.contactPersonId,
      required this.extraInfo});

  factory Club.fromJson(Map<String, dynamic> json) {
    return Club(
      id: json['id'],
      name: json['name'],
      streetAddress: json['streetAddress'],
      streetNumber: json['streetNumber'],
      zipCode: json['zipCode'],
      phoneNumber: json['phoneNumber'],
      contactPersonId: json['contactPersonId'],
      extraInfo: json['extraInfo'],
    );
  }
}

File: go/internal/services/club.go
package services

import (
	"context"
	"errors"
)

var (
	ErrorNotImplemented = errors.New("Error method is not implemented")
	ErrorNotFound       = errors.New("Error item not found")
	ErrorNoAccess       = errors.New("Error user does not have access to item")
)

type Club struct {
	Id              int    `json:"id"`
	Name            string `json:"name"`
	StreetAddress   string `json:"streetAddress"`
	StreetNumber    string `json:"streetNumber"`
	ZipCode         string `json:"zipCode"`
	PhoneNumber     string `json:"phoneNumber"`
	ContactPersonId int    `json:"contactPersonId"`
	ExtraInfo       string `json:"extraInfo"`
}

// ClubStore - this interface defines all methods that our service needs
// to manipulate the storage of clubs
type ClubStore interface {
	GetAllClub(context.Context) ([]Club, error)
	GetClub(context.Context, int) (Club, error)
	UpdateClub(ctx context.Context, id int, dc Club) (Club, error)
	DeleteClub(ctx context.Context, id int) error
	CreateClub(ctx context.Context, dc Club) (Club, error)
}

type ClubService struct {
	ClubStore ClubStore
}

func NewClubService(clubStore ClubStore) *ClubService {
	return &ClubService{
		ClubStore: clubStore,
	}
}

func (service *ClubService) GetAllClub(ctx context.Context) ([]Club, error) {
	dcs, err := service.ClubStore.GetAllClub(ctx)
	if err != nil {
		return []Club{}, err
	}
	return dcs, nil
}

func (service *ClubService) GetClub(ctx context.Context, id int) (Club, error) {
	dc, err := service.ClubStore.GetClub(ctx, id)
	if err != nil {
		return Club{}, err
	}
	return dc, nil
}

func (service *ClubService) UpdateClub(ctx context.Context, id int, dc Club) (Club, error) {
	return service.ClubStore.UpdateClub(ctx, id, dc)
}

func (service *ClubService) DeleteClub(ctx context.Context, id int) error {
	return service.ClubStore.DeleteClub(ctx, id)
}

func (service *ClubService) CreateClub(ctx context.Context, dc Club) (Club, error) {
	dc, err := service.ClubStore.CreateClub(ctx, dc)
	if err != nil {
		return Club{}, err
	}
	return dc, nil
}
